server/usermanager.ts

```diff
@@ -744,28 +750,20 @@ const accountRecoveryStart: RequestHandler<
 	OttResponseBody<{}>,
 	OttApiRequestAccountRecoveryStart
 > = async (req, res) => {
+	const body = OttApiRequestAccountRecoveryStartSchema.parse(req.body);
 	if (conf.get("rate_limit.enabled")) {
 		await consumeRateLimitPoints(res, req.ip, 100);
 	}
 
-	if (!("email" in req.body) && !("username" in req.body)) {
-		throw new BadApiArgumentException(
-			"email,username",
-			"Either email or username is required."
-		);
-	}
-
 	const query = {
-		user: req.body.email ?? req.body.username,
+		user: (body as { email?: string }).email ?? (body as { username?: string })?.username,
```

Comment:
While this technically works, it wont throw type errors if the type of `body` changes because `as` forces a type override. Take a look at the type narrowing docs: https://www.typescriptlang.org/docs/handbook/2/narrowing.html